fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -3010,6 +3013,141 @@ public CommandProcessingResult makeRefund(final Long loanId, final LoanTransacti
                 .build();
     }
 
+    @Transactional
+    @Override
+    public CommandProcessingResult makeManualInterestRefund(final Long loanId, final Long transactionId, final JsonCommand command) {
+        loanTransactionValidator.validateManualInterestRefundTransaction(command.json());
+        Loan loan = loanAssembler.assembleFrom(loanId);
+        if (loan == null) {
+            throw new LoanNotFoundException(loanId);
+        }
+        final LoanTransaction targetTransaction = loan.getLoanTransaction(txn -> txn.getId() != null && txn.getId().equals(transactionId));
```

Comment:
Please fetch this from database... we are moving away from `loan.getLoanTransaction`